qutebrowser/completion/models/listcategory.py

```diff
@@ -63,6 +63,9 @@ def set_pattern(self, val):
         val = re.sub(r' +', r' ', val)  # See #1919
         val = re.escape(val)
         val = val.replace(r'\ ', '.*')
+        if len(val) > 5000:  # avoid crash on huge search terms (#5973)
+            log.completion.warning("Pattern too long ({})".format(len(val)))
+            val = "^$"
```

Comment:
I wonder, should it be ignore entirely like here, or should we just cut the pattern off instead? Though I suppose we'd need to trim the original input rather than the generated pattern, so that we don't cut things off in the middle after a `\` or so somewhere.

Reply:
That might be nice, but trimming based on the generated pattern would be tricky. We'd either need to count substitutions or guess-and-check, which seems like too much work for such an edge case.

Reply:
Does it matter much with a conservative limit, though? If we assume the actual limit is 50k but our limit is 5k, it really doesn't matter if the escaped pattern gets longer (in the extreme case, I guess it'd  be doubled at 10k characters).

Reply:
Oh, that's right! :facepalm: I'll do that